runtime/runtime.c

```diff
@@ -289,28 +21,39 @@ __attribute__((noreturn)) void awsm_abi__trap_unreachable() {
 WEAK u32 wasmg___heap_base = 0;
 u32      runtime_heap_base;
 
-u32 allocate_n_bytes(u32 n) {
-    awsm_assert(memory_size > 0);
-    u32 res = runtime_heap_base;
-    runtime_heap_base += n;
-    while (memory_size < runtime_heap_base) {
-        expand_memory();
+// If a function is registered using the (start) instruction, call it
+WEAK void awsm_abi__start_fn() {}
+
+#ifdef DYNAMIC_LINKING_WASM_SO
```

Comment:
Is the only way to do this with `ifdef`s? I thought we tried to avoid them in awsm as much as possible. Could be misremembering.

Reply:
Given your later code, is it accurate to say that this is the *complete* set of symbols that an environment depends on an awsm module for?

I don't think it captures all of the symbols that a module depends on an environment for, right?

Reply:
Got rid of the ifdef everywhere...

Reply:
ended up separating the wasm instructions for all the architectures in the repo  (mpx, cortex-m ...)